patches/tock/09-add-vendor-hid-usb-interface.patch

```diff
@@ -0,0 +1,331 @@
++                            hil::usb::CtrlSetupResult::Ok
++                        } else {
++                            hil::usb::CtrlSetupResult::ErrGeneric
++                        }
+                     } else {
+                         hil::usb::CtrlSetupResult::ErrGeneric
+                     }
+diff --git a/capsules/src/usb/usbc_ctap_hid.rs b/capsules/src/usb/usbc_ctap_hid.rs
+index 642039120..53fbc7719 100644
+--- a/capsules/src/usb/usbc_ctap_hid.rs
++++ b/capsules/src/usb/usbc_ctap_hid.rs
+@@ -44,21 +44,59 @@ static CTAP_REPORT_DESCRIPTOR: &'static [u8] = &[
+     0xC0, // HID_EndCollection
+ ];
+ 
++#[cfg(feature = "vendor_hid")]
++static VENDOR_REPORT_DESCRIPTOR: &'static [u8] = &[
++    0x06, 0x00, 0xFF, // HID_UsagePage ( VENDOR ),
++    0x09, 0x00, // HID_Usage ( Unused ),
++    0xA1, 0x01, // HID_Collection ( HID_Application ),
++    0x09, 0x20, // HID_Usage ( FIDO_USAGE_DATA_IN ),
++    0x15, 0x00, // HID_LogicalMin ( 0 ),
++    0x26, 0xFF, 0x00, // HID_LogicalMaxS ( 0xff ),
++    0x75, 0x08, // HID_ReportSize ( 8 ),
++    0x95, 0x40, // HID_ReportCount ( HID_INPUT_REPORT_BYTES ),
++    0x81, 0x02, // HID_Input ( HID_Data | HID_Absolute | HID_Variable ),
++    0x09, 0x21, // HID_Usage ( FIDO_USAGE_DATA_OUT ),
++    0x15, 0x00, // HID_LogicalMin ( 0 ),
++    0x26, 0xFF, 0x00, // HID_LogicalMaxS ( 0xff ),
++    0x75, 0x08, // HID_ReportSize ( 8 ),
++    0x95, 0x40, // HID_ReportCount ( HID_OUTPUT_REPORT_BYTES ),
++    0x91, 0x02, // HID_Output ( HID_Data | HID_Absolute | HID_Variable ),
++    0xC0, // HID_EndCollection
++];
++
+ static CTAP_REPORT: ReportDescriptor<'static> = ReportDescriptor {
+     desc: CTAP_REPORT_DESCRIPTOR,
+ };
+ 
++#[cfg(feature = "vendor_hid")]
++static VENDOR_REPORT: ReportDescriptor<'static> = ReportDescriptor {
++  desc: VENDOR_REPORT_DESCRIPTOR,
++};
++
+ static HID_SUB_DESCRIPTORS: &'static [HIDSubordinateDescriptor] = &[HIDSubordinateDescriptor {
+     typ: DescriptorType::Report,
+     len: CTAP_REPORT_DESCRIPTOR.len() as u16,
+ }];
+ 
++#[cfg(feature = "vendor_hid")]
++static VENDOR_HID_SUB_DESCRIPTORS: &'static [HIDSubordinateDescriptor] = &[HIDSubordinateDescriptor {
++  typ: DescriptorType::Report,
++  len: VENDOR_REPORT_DESCRIPTOR.len() as u16,
++}];
++
+ static HID: HIDDescriptor<'static> = HIDDescriptor {
+     hid_class: 0x0110,
+     country_code: HIDCountryCode::NotSupported,
+     sub_descriptors: HID_SUB_DESCRIPTORS,
+ };
+ 
++#[cfg(feature = "vendor_hid")]
++static VENDOR_HID: HIDDescriptor<'static> = HIDDescriptor {
++  hid_class: 0x0110,
++  country_code: HIDCountryCode::NotSupported,
++  sub_descriptors: VENDOR_HID_SUB_DESCRIPTORS,
++};
++
+ pub struct ClientCtapHID<'a, 'b, C: 'a> {
+     client_ctrl: ClientCtrl<'a, 'static, C>,
+ 
+@@ -82,6 +120,10 @@ impl<'a, 'b, C: hil::usb::UsbController<'a>> ClientCtapHID<'a, 'b, C> {
+         product_id: u16,
+         strings: &'static [&'static str],
+     ) -> Self {
++        if cfg!(feature = "vendor_hid") {
```

Comment:
Why use the other cfg style here, not 
`#[cfg(feature = "vendor_hid")]`
?

Reply:
Most likely because the else branch was printing something else during development. That is not needed now, so changed to other style.